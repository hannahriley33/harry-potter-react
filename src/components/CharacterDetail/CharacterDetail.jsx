import React from 'react';
import PropTypes from 'prop-types';

const CharacterDetail = ({ image, name, house, patronus }) => (
  <section>
    <img src={image} />
    <h2>{name}</h2>
    <p>Hogwarts House: {house}</p>
    <p>Patronus Form: {patronus}</p>
  </section>
);

CharacterDetail.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  house: PropTypes.string.isRequired,
  patronus: PropTypes.string.isRequired
};

export default CharacterDetail;
