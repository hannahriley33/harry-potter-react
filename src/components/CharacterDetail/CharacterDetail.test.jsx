import React from 'react';
import { shallow } from 'enzyme';
import CharacterDetail from './CharacterDetail.jsx';

describe('Character Detail', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<CharacterDetail image="abc" name="harry" house="gryffindor"
      patronus="stag" />);
    expect(wrapper).toMatchSnapshot();
  });
});


