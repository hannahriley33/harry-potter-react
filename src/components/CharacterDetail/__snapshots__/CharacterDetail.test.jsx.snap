// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Detail matches the snapshot 1`] = `
<section>
  <img
    src="abc"
  />
  <h2>
    harry
  </h2>
  <p>
    Hogwarts House: 
    gryffindor
  </p>
  <p>
    Patronus Form: 
    stag
  </p>
</section>
`;
